### PR TITLE
Reduce thread contention between main thread and asset fetch threads

### PR DESCRIPTION
**Pull request by RyeMutt**
Reduce thread contention between the main thread and all asset fetch threads due to frequent access to the singleton master dependency list by utilizing simpletons and statics.

Found while profiling:
image

### CODE REVIEW

File: indra/llfilesystem/llfilesystem.cpp [hunk 1]
```diff
@@ -325,3 +320,64 @@ bool LLFileSystem::remove()
 
     return true;
 }
+
+void LLFileSystem::updateFileAccessTime(const std::string file_path)
```
Comment (Ansariel) [inline indra/llfilesystem/llfilesystem.cpp:324]: Forgot to also change it to a const ref or even string_view?

Comment (RyeMutt) [reply indra/llfilesystem/llfilesystem.cpp]: Good catch! Fixed!

File: indra/llfilesystem/lldiskcache.cpp [hunk 2]
```diff
@@ -236,97 +246,17 @@ const std::string LLDiskCache::assetTypeToString(LLAssetType::EType at)
...
-    {
-        boost::filesystem::last_write_time(file_path, cur_time, ec);
-    }
-#endif
-
-    if (ec.failed())
-    {
-        LL_WARNS() << "Failed to update last write time for cache file " << file_path << ": " << ec.message() << LL_ENDL;
-    }
+    return llformat("%s%s%s_%s_0.asset", sCacheDir.c_str(), gDirUtilp->getDirDelimiter().c_str(), CACHE_FILENAME_PREFIX.c_str(), id.c_str());
```
Comment (Ansariel) [inline indra/llfilesystem/lldiskcache.cpp:251]: Could probably omit the "_0" entirely and increase the cache version to trigger a cache clear.

**Review by marchcat — APPROVED** [review]
Thank you!

**Comment by RyeMutt** [comment]
Build error on mac should be fixed now.

**Comment by RyeMutt** [comment]
Windows failure appears to be CI-script related

File: indra/llfilesystem/lldiskcache.cpp (hunk 2 above)
Comment (RunitaiLinden) [reply indra/llfilesystem/lldiskcache.cpp:251]: If I'm reading the code right, that _0 was always there and "extra_info" was always empty, so _0 is just the status quo.

**Pull request merged**